docs/rfcs/016-user-and-database-management.md

```diff
@@ -0,0 +1,68 @@
+## Zentih user and database management RFC
+
+### CREATE USER
+
+CREATE USER is only allowed from console, 
+because it requires generation of the access token and other complex logic.
+
+Client has to chose which databases will be accessible by the new user.
+Along with database access, we explicitly grant CREATE ON DATABASE privilege to the user.
+`GRANT CREATE ON DATABASE .. TO .. ;`
+to let the user to create extensions.
+
+All new users are granted with CREATEDB privilege by default.
+`CREATE ROLE .. WITH CREATEDB`
+Later we may add an option to manage it.
+
+*Q: Are we going to allow user to create new users and NOLIGIN roles to manage privileges?
+*`CREATE ROLE .. WITH CREATEROLE`
+
+
+### CREATE DATABASE
+
+Client can CREATE DATABASE via SQL as well as via console.
+
+When database is created from SQL (not by superuser)
+we intercept this action using DDL hook and send it to the console to synchronize the information about databases.
+
+### CREATE EXTENSION
+
+To create trusted extensions in the database, user must have an access privileges on the database.
+
+We explicitly grant CREATE ON DATABASE privilege to the new user along with database access.
+If user creates a database, he becomes the owner and thus has CREATE ON DATABASE privilege by default.
+
+### DROP EXTENSION
+
+Nothing special. PostgreSQL behavior.
+
+### DROP DATABASE
+
+From PostgreSQL docs:
+The right to drop an object, or to alter its definition in any way, is not treated as a grantable privilege; it is inherent in the owner, and cannot be granted or revoked. (However, a similar effect can be obtained by granting or revoking membership in the role that owns the object) 
+
+Note that ownership can be changed using `REASSIGN OWNED BY .. TO ..` 
+
+### DROP USER
+
+DROP USER is only allowed from console.
+
+In case some objects depend on the user, PostgreSQL throws a detailed error.
+
+- ERROR:  role "p" cannot be dropped because some objects depend on it
+DETAIL:  owner of database pdb
+1 object in database pdb
+
+### Competitiors
+
+Competing databases have more or less standart approach to user management.
```

Comment:
Have you tried how they actually behave? Are there a list of dbs / users in UI? What happens when you create user in shell? Is it immediately updated in UI?

I've played a bit with CockroachDB serverless cloud. They are weird :)

1) You can create both dbs and roles via shell and UI
2) If you create a role via psql you can +- immediately connect using it
3) **But** it never (in some reasonable time) appears in the UI, so it seems like never synced. If you try to create role with the same name in UI it shows "already exists" error, but it's still not listed
4) Yet, dbs created via psql are listed in UI. Considering the load time it refreshes them rather frequently

I'm going to try Planetscale as well

Reply:
Hm, Planetscale seems like a completely different thing %) First of all it's MySQL (yeah, I'm a slowpoke). Second, they don't allow any roles management (as far as I discovered) with a very strange error:
```sql
mysql> create user 'test';
ERROR 1105 (HY000): syntax error at position 12 near 'user'
mysql> create table t(i int);
ERROR 1105 (HY000): direct DDL is disabled
mysql> create role 'test';
ERROR 1105 (HY000): syntax error at position 12 near 'role'
```
According to the docs syntax is correct
https://dev.mysql.com/doc/refman/8.0/en/create-role.html
https://dev.mysql.com/doc/refman/8.0/en/create-user.html

Reply:
YandexCloud PostgreSQL only allows to create users and databases via their Console (CLI).
There are two "admin" roles but they have very restricted privileges too: https://cloud.yandex.ru/docs/managed-postgresql/concepts/roles

To install extensions, the user should choose them on the Database page in Console.

Reply:
> Planetscale

That's not PostgreSQL.

> YandexCloud

Is that competing? Either way, if we're going to provide "PostgreSQL as a service", I don't expect some bastardized variant that requires a non-PostgreSQL API for doing tasks generally done through the PostgreSQL client, but a fully featured variant that provides all possible PostgreSQL features through the PostgreSQL connection.

Reply:
> That's not PostgreSQL

Do I write somewhere that it **is** PostgreSQL? I don't see any reason to don't try some other relational cloud DBaaS.